File: mvp/python/adapters/sink/http_push.py
import json, urllib.request

class HttpSink:
    def __init__(self, cfg):
        self.url = cfg['url']
    def send(self, payload: dict):
        data = json.dumps(payload).encode('utf-8')
        req = urllib.request.Request(self.url, data=data, headers={'Content-Type':'application/json'})
        with urllib.request.urlopen(req, timeout=5) as resp:
            return resp.status

File: mvp/ts/src/adapters/sink/http_push.ts
export class HttpSink {
  url: string;
  constructor(cfg:any){ this.url = cfg.url; }
  async send(payload: any){
    // Using undici would be typical; keep Node fetch for simplicity if enabled
    const res = await fetch(this.url, { method:"POST", headers:{ "Content-Type":"application/json" }, body: JSON.stringify(payload) });
    return res.status;
  }
}
